### PR TITLE
feat: add a simple redux counter app

### PR DESCRIPTION
**Pull request by igor-ka**
@SethDavenport I removed the old flux ngcourse-next based application and put a simple redux counter application instead.

Next step would be to port the react-redux-seed application in here

### CODE REVIEW

File: src/components/counter/counter.ts [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  return {
+    restrict: 'E',
+    controllerAs: 'ctrl',
+    controller: CounterController,
+    template: require('./counter.html'),
+    scope: {}
+  };
+}
+
+class CounterController {
```
Comment (e-schultz) [inline src/components/counter/counter.ts]: In sad-ui, we would have the controller be in it's own JS file outside of the directive-deffinition object, made it a bit easier to unit test as we could just 

```javascript
import controller from './controller-file`
// later on in the tests

let ctrl = new Controller(my,mocks) { 
}
```

Suppose we can still do that here, but it'd be more

```javascript
import {CounterComponent} from './counter';
// later on in the test, something like ??
let ctrl = new CounterComponent().Controller(my,mocks)
```

or, could also just have an export class CounterController here.

On the other hand - when controllers are being this thin - having unit tests here are more just 'upping the lines of coverage', but not adding much value.

Thoughts?

File: src/index.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+const thunk = require('redux-thunk');
+
+angular.module('counter', [ngRedux])
+  .config([
+    '$ngReduxProvider',
+    ($ngReduxProvider) => 
+      $ngReduxProvider.createStoreWith(rootReducer, [thunk])])
+  .directive('ngrCounter', CounterComponent);
+
+angular.element(document).ready(
```
Comment (e-schultz) [inline src/index.ts]: IMO we should also include the logger middleware. It's very handy - and something I can see most people wanting to add anyways.

File: src/components/counter/counter.ts (hunk 1 above)
Comment (igor-ka) [inline src/components/counter/counter.ts]: hey @e-schultz the rationale to keep it all in one file was to align better with Angular 2.0. The way I envision this is that the controller will end up pretty much the same, while the DDO will be "converted" to Angular 2.0 annotations.

Other than that I have no preferences over exporting the controller, or using the .Controller(my...) syntax shown above.

Also, :+1: for the unit tests coverage bumping note, couldn't agree more.

File: src/index.ts (hunk 2 above)
Comment (igor-ka) [inline src/index.ts]: @e-schultz definitely.

I've added this PR just to get started for now, and chip away at it in the future.

**Comment by e-schultz** [comment]
I'm fine to :shipit: the other tweaks/etc can come in as other PRs - but this gives it a good start.

**Pull request merged**